api/src/main/java/org/pmiops/workbench/exfiltration/jirahandler/EgressJiraHandler.java

```diff
@@ -0,0 +1,97 @@
+package org.pmiops.workbench.exfiltration.jirahandler;
+
+import com.google.common.collect.ImmutableMap;
+import java.time.Clock;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+import org.pmiops.workbench.db.model.DbEgressEvent;
+import org.pmiops.workbench.exfiltration.EgressRemediationAction;
+import org.pmiops.workbench.jira.ApiException;
+import org.pmiops.workbench.jira.JiraContent;
+import org.pmiops.workbench.jira.JiraService;
+import org.pmiops.workbench.jira.JiraService.IssueProperty;
+import org.pmiops.workbench.jira.JiraService.IssueType;
+import org.pmiops.workbench.jira.model.AtlassianContent;
+import org.pmiops.workbench.jira.model.CreatedIssue;
+import org.pmiops.workbench.jira.model.IssueBean;
+import org.pmiops.workbench.jira.model.SearchResults;
+import org.springframework.stereotype.Service;
+
+@Service
+public abstract class EgressJiraHandler {
+
+  private static final Logger log = Logger.getLogger(EgressJiraHandler.class.getName());
+
+  private final Clock clock;
+  private final JiraService jiraService;
+
+  protected EgressJiraHandler(Clock clock, JiraService jiraService) {
+    this.clock = clock;
+    this.jiraService = jiraService;
+  }
+
+  public abstract void logEventToJira(DbEgressEvent event, EgressRemediationAction action)
+      throws ApiException;
+
+  abstract Stream<AtlassianContent> jiraEventDescription(
+      DbEgressEvent event, EgressRemediationAction action);
+
+  abstract Stream<AtlassianContent> jiraEventComment(
+      DbEgressEvent event, EgressRemediationAction action);
+
+  protected SearchResults searchJiraIssuesWithLabel(
+      DbEgressEvent event, String envShortName, String label) throws ApiException {
+    return jiraService.searchIssues(
+        // Ideally we would use Resolution = Unresolved here, but due to a misconfiguration of
+        // RW Jira, transitioning to Won't Fix / Duplicate do not currently resolve an issue.
+        String.format(
```

Comment:
I added LABELS here to the search criteria to differentiate between the regular high egress and the file lengths high egress. Look at this test task which contained a different label
[RW-8937](https://precisionmedicineinitiative.atlassian.net/browse/RW-8937)